spec/features/stripe_checkout_spec.rb

```diff
@@ -45,52 +46,82 @@
+      find_field("Expiration").send_keys(date)
+    end
+
+    allow_any_instance_of(Spree::Order).to receive(:for_delivery?).and_return(true)
   end
 
   # This will fetch a token from Stripe.com and then pass that to the webserver.
   # The server then processes the payment using that token.
-  it "can process a valid payment", js: true do
-    fill_in "Card Number", with: "4242 4242 4242 4242"
-    fill_in "Card Code", with: "123"
-    fill_in "Expiration", with: "01 / #{Time.now.year + 1}"
-    click_button "Save and Continue"
-    expect(page).to have_current_path("/checkout/confirm")
-    click_button "Place Order"
-    expect(page).to have_content("Your order has been processed successfully")
-  end
+  context "when the CC number is valid" do
+    let(:cc_number) { "4242 4242 4242 4242" }
 
-  it "shows an error with a missing credit card number", js: true do
-    fill_in "Expiration", with: "01 / #{Time.now.year + 1}"
-    click_button "Save and Continue"
-    expect(page).to have_content("Could not find payment information")
-  end
+    context "when the expiration is valid" do
+      let(:cc_expiration) { "01 / #{Time.current.year + 1}" }
 
-  it "shows an error with a missing expiration date", js: true do
-    fill_in "Card Number", with: "4242 4242 4242 4242"
-    click_button "Save and Continue"
-    expect(page).to have_content("Your card's expiration year is invalid.")
-  end
+      it "processes a valid payment", js: true do
+        fill_in "Card Code", with: "123"
+        click_button "Save and Continue"
+        expect(page).to have_current_path("/checkout/confirm")
+        click_button "Place Order"
+        expect(page).to have_content("Your order has been processed successfully")
+      end
 
-  it "shows an error with an invalid credit card number", js: true do
-    fill_in "Card Number", with: "1111 1111 1111 1111"
-    fill_in "Expiration", with: "01 / #{Time.now.year + 1}"
-    click_button "Save and Continue"
-    expect(page).to have_content("Your card number is incorrect.")
+      context "when the security fields are invalid" do
+        let(:cc_security) { "12" }
+
+        it "shows an error", js: true do
+          fill_in "Card Code", with: cc_security
+          click_button "Save and Continue"
+          expect(page).to have_content("Your card's security code is invalid.")
+        end
+      end
+    end
+
+    context "when the expiration fields are invalid" do
+      let(:cc_expiration) { "00 / #{Time.current.year + 1}" }
+
+      it "shows an error", js: true do
+        fill_in "Card Code", with: "123"
+        click_button "Save and Continue"
+        expect(page).to have_content("Your card's expiration month is invalid.")
+      end
+    end
+
+    context "when the expiration fields are missing" do
+      let(:cc_expiration) { "" }
+
+      it "shows an error", js: true do
+        click_button "Save and Continue"
+        expect(page).to have_content("Your card's expiration year is invalid.")
```

Comment:
interesting that when the expiration month/year is missing...it complains about the year first